Settings Redesign

benisgold opened the pull request:
Fixes TEAM2-247, TEAM2-78, TEAM2-77, TEAM2-79

Figma link (if any):
https://www.figma.com/file/Xc4Pq3CLJNVG52z1K0URvP/notifications?node-id=647:4924

## What changed (plus any additional context for devs)
- settings redesign on all settings pages
- typescript everything
- foundation for notification settings
- replaced all existing components with design system equivalents, including screens that weren't explicitly redesigned

## Screen recordings / screenshots
https://www.loom.com/share/ec325c353a5743ce9f00ecc06d7b6879


## What to test
- make sure all settings screens are still there
- make sure no functionality has changed
- make sure the implementation matches the figma designs (see above)
- go to `src/components/settings-menu/SettingsSection.tsx` and enable + test all the settings that may be hidden for various reasons. should be as easy as replacing something like `{conditional + <Component />}` with `{true + <Component />}`


## Final checklist

- [ ] Assigned individual reviewers?
- [ ] Added labels? (`team1/team2`, `critical path`, `release`, `dev QA`)
- [ ] Did you test both iOS and Android?
- [ ] If your changes are visual, did you check both the light and dark themes?
- [ ] Added e2e tests? If not, please specify why
- [ ] If you added new critical path files, did you update the CODEOWNERS file?
- [ ] If no `dev QA` label, did you add the PR to the QA Queue?

pugson commented on src/components/settings-menu/components/MenuItem.tsx:
```diff
@@ -0,0 +1,219 @@
+    size="18px"
+    weight={weight}
+  >
+    {text}
```
<img width="344" alt="image" src="https://user-images.githubusercontent.com/6843656/180036761-74e372d0-e36a-4142-838a-ba2fd79a98cd.png">

pugson replied:
one small nit here, this ends up moving to the second line instead of the text breaking into 2 lines and being vertically centered with the emoji.

tried to fix it but couldn't figure out how so maybe you know

pugson reviewed (approved):
looking hot 🔥🔥🔥

jxom commented:
/testflight

jxom commented:
/testflight

jkadamczyk commented:
Code looks super well!
Issues found:
- The statusbar color is not changing when you open the settings modal and the background is going dark. Would make sense to make the statusbar color change when opening the settings modal and go back to previous when closing it. I think it may be a regression since the prod version changes the status bar theme

jkadamczyk reviewed (commented):
Code is cool, mentioned found bug in the comment above

jxom commented:
/testflight

benisgold commented on src/components/settings-menu/NotificationsSection.tsx:
```diff
@@ -0,0 +1,121 @@
+              value="notben.eth"
+            />
+          }
+          onPress={() => onPress('notben.eth')}
```
these are just placeholders for future pr's, users won't see this since this menu will be locked behind the notifications feature flag for now

estebanmino replied:
is this still the case?

benisgold commented:
@jkadamczyk do you mean you see a white status bar? i see a black status bar, the same as develop

jxom reviewed (commented):
Looking great so far! I have a few comments.

benisgold commented on src/components/settings-menu/components/Menu.tsx:
```diff
@@ -0,0 +1,49 @@
+          <Stack separator={<Divider color="divider60" />}>{children}</Stack>
+        </Box>
+        {!!description && (
+          <Box paddingHorizontal={{ custom: 16 }} paddingTop="5px">
```
yeah bc the first two components only need a 12px space but these two need a 17px space.  i got rid of the stack entirely and just used paddingTop and paddingBottom for the corresponding components

benisgold commented on src/components/settings-menu/components/MenuItem.tsx:
```diff
@@ -0,0 +1,219 @@
+  titleComponent: React.ReactNode;
+  labelComponent?: React.ReactNode;
+  disabled?: boolean;
+  hasChevron?: boolean;
```
one place: the Theme menu item

benisgold commented on src/components/settings-menu/components/MenuItem.tsx:
```diff
@@ -0,0 +1,219 @@
+  rightComponent?: React.ReactNode;
+  leftComponent?: React.ReactNode;
+  size: 'medium' | 'large';
+  iconPadding?: 'small' | 'medium' | 'large';
```
damn u right

benisgold replied:
gigabrain

benisgold commented on src/components/settings-menu/components/MenuItem.tsx:
```diff
@@ -0,0 +1,219 @@
+      color={color}
+      colors={colors}
+      fillColor={colors.white}
+      shadowColor={isDarkMode ? colors.shadow : color}
```
can i specify a shadow color in the design system?

benisgold replied:
I tried this but it didn't work

`shadow={
        {
          android: {
            color: isDarkMode ? colors.shadow : statusColors[status],
            elevation: 24,
            opacity: 0.8,
          },
          ios: [
            {
              blur: 12,
              color: isDarkMode ? colors.shadow : statusColors[status],
              offset: { x: 0, y: 4 },
              opacity: 0.3,
            },
            {
              blur: 24,
              color: isDarkMode ? colors.shadow : statusColors[status],
              offset: { x: 0, y: 8 },
              opacity: 0.15,
            },
          ],
        } as ShadowValue
      }
`

benisgold commented on src/styles/colors.ts:
```diff
@@ -185,6 +185,11 @@ const getColorsByTheme = (darkMode?: boolean) => {
     ropsten: '#ff4a8d',
   };
 
+  let settings = {
```
ran it by him, going with it!

benisgold commented:
/testflight

brunobar79 commented:
CI failing

pugson commented:
rebase required 😭

benisgold commented:
/testflight

BrodyHughes reviewed (changes requested):
A few errors I see atm:

This button width:
<img width="368" alt="Screen Shot 2022-08-12 at 10 29 06 AM" src="https://user-images.githubusercontent.com/41711440/184389014-a0f1990d-a7d4-4636-99ec-49bd3b8d34bc.png">

Do we want the Network button to function like the themes button or vice versa? Seems odd to have two buttons with similar functionality and very different execution right next to eachother (networks isn't on the figma design doc):
<img width="369" alt="Screen Shot 2022-08-12 at 10 30 27 AM" src="https://user-images.githubusercontent.com/41711440/184389021-bdb31116-6093-4a9a-a2ec-0c3e39664961.png">
<img width="370" alt="Screen Shot 2022-08-12 at 10 30 35 AM" src="https://user-images.githubusercontent.com/41711440/184389022-a699aba8-97a4-4526-b87c-6388755bd4f7.png">
<img width="358" alt="Screen Shot 2022-08-12 at 10 31 11 AM" src="https://user-images.githubusercontent.com/41711440/184389024-e9270392-0a99-4e9b-b04c-4f34139fb00e.png">
<img width="368" alt="Screen Shot 2022-08-12 at 10 31 49 AM" src="https://user-images.githubusercontent.com/41711440/184389027-9c5bf2f4-49df-4399-8908-84b7cdeb7d79.png">

benisgold commented:
thanks for the qa @BrodyHughes !!

1. randomly large button: i think i fixed this in a very recent commit (so understandable why you might have missed it) but i will double check.
2. design stuff: i went over it thoroughly with Jeremy a few times over so I think we're probably set on that front
3. notifications menu: this is intentionally just a skeleton, pugson is working on the functionality. it's hidden behind the notifications feature flag and isn't user-facing.
4. missing context menu: i implemented that in a separate pr that i planned on merging after, but i think i'll just include that here

will work on these action items and get back to you when they are complete

BrodyHughes commented:
absolutely killing it 🥇

benisgold commented:
![Simulator Screen Shot - iPhone 13 Pro - 2022-08-12 at 10 35 19](https://user-images.githubusercontent.com/15272675/184403835-d3360cb8-da09-403f-96d7-11097ebffb0a.png)

benisgold commented:
@BrodyHughes 

here's an edit wallet context menu demo

https://www.loom.com/share/9075603e7b4e41d8b6d1e8a9b97fb9f5

benisgold commented:
/testflight

benisgold commented:
/testflight

estebanmino commented on src/components/secret-display/SecretDisplaySection.tsx:
```diff
@@ -13,98 +13,49 @@ import Spinner from '../Spinner';
   const [visible, setVisible] = useState(true);
   const [isRecoveryPhraseVisible, setIsRecoveryPhraseVisible] = useState(false);
-  const [seed, setSeed] = useState(null);
+  const [seed, setSeed] = useState(null as any);
```
is the `as any` needed? maybe you want `useState<string | null>(null)` ?

benisgold replied:
i just wanted the ts warning to go away. i didn't know about this way to do it which looks way better

estebanmino commented on src/components/secret-display/SecretDisplaySection.tsx:
```diff
@@ -148,68 +99,81 @@ export default function SecretDisplaySection({
           face identification. To see your recovery phrase, turn on biometrics
           in your phone’s settings.
-        </BiometryWarningText>
+        </Text>
```
do you want to add the above text to languages files?

estebanmino commented on src/components/settings-menu/BackupSection/AlreadyBackedUpView.tsx:
```diff
@@ -197,25 +152,30 @@ export default function AlreadyBackedUpView() {
+    >
+      <Box marginTop="-10px">
+        <Text color="secondary50" size="14px" weight="medium">
+          {(walletStatus === WalletBackupStatus.CLOUD_BACKUP && `Backed up`) ||
```
do you want to add the text to languages files?

estebanmino commented on src/components/settings-menu/BackupSection/AlreadyBackedUpView.tsx:
```diff
@@ -197,25 +152,30 @@ export default function AlreadyBackedUpView() {
+        <Text color="secondary50" size="14px" weight="medium">
+          {(walletStatus === WalletBackupStatus.CLOUD_BACKUP && `Backed up`) ||
+            (walletStatus === WalletBackupStatus.MANUAL_BACKUP &&
+              `Backed up manually`) ||
```
do you want to add the text to languages files?

estebanmino commented on src/components/settings-menu/BackupSection/AlreadyBackedUpView.tsx:
```diff
@@ -197,25 +152,30 @@ export default function AlreadyBackedUpView() {
+          {(walletStatus === WalletBackupStatus.CLOUD_BACKUP && `Backed up`) ||
+            (walletStatus === WalletBackupStatus.MANUAL_BACKUP &&
+              `Backed up manually`) ||
+            (walletStatus === WalletBackupStatus.IMPORTED && `Imported`)}
```
do you want to add the text to languages files?

benisgold commented:
TF 1.7.2 (6)

estebanmino commented on src/components/settings-menu/BackupSection/AlreadyBackedUpView.tsx:
```diff
@@ -197,25 +152,30 @@ export default function AlreadyBackedUpView() {
+        <Text size="20px" weight="bold">
+          {(walletStatus === WalletBackupStatus.IMPORTED &&
+            `Your wallet was imported`) ||
+            `Your wallet is backed up`}
```
do you want to add the text to languages files?

estebanmino commented on src/components/settings-menu/BackupSection/BackupSection.tsx:
```diff
@@ -0,0 +1,156 @@
+        {Object.keys(wallets)
+          .filter(key => wallets[key].type !== WalletTypes.readOnly)
+          .map(key => {
+            const wallet = wallets[key];
+            const visibleAccounts = wallet.addresses.filter(
+              (a: any) => a.visible
+            );
+            const account = visibleAccounts[0];
+            const totalAccounts = visibleAccounts.length;
+            const { color, label, address } = account;
+            if (wallet.backupType === WalletBackupTypes.cloud) {
+              cloudBackedUpWallets += 1;
+            }
+            let labelOrName = label;
+            if (!label) {
+              if (walletNames[address]) {
+                labelOrName = walletNames[address];
+              }
+            }
+
+            return (
```
wondering if we should move this logic out of `return`, feels strange to have this logic (and logic below this comment) together with rendered components, wdyt?

estebanmino commented on src/components/settings-menu/components/MenuContainer.tsx:
```diff
@@ -0,0 +1,23 @@
+  return (
+    <ScrollView scrollEventThrottle={32} testID={testID}>
+      <Box
+        paddingBottom={{ custom: 34.5 }}
```
34.5 feels very custom, this was on designs?

benisgold replied:
there was definitely a reason for this... let me think lol

benisgold replied:
<img width="220" alt="Screen Shot 2022-08-15 at 4 31 03 PM" src="https://user-images.githubusercontent.com/15272675/184732023-e48047f8-527a-4e49-a1a6-d465822c5d94.png">

i wanted to replicate the implicit top padding here

benisgold replied:
ended up reworking `MenuContainer` a bit in ios to improve the scroll view, so the above response is obsolte

estebanmino commented on src/styles/colors.ts:
```diff
@@ -106,7 +106,7 @@ const getColorsByTheme = (darkMode?: boolean) => {
     green: '#2CCC00', // '58, 166, 134'
     grey: '#A9ADB9', // '169, 173, 185'
-    grey20: '#333333', // '51, 51, 51'
+    grey20: '#333333', // '51, 51, 51',
```
last `,` was on purpose?

estebanmino reviewed (commented):
Sync code push row is not aligned on android dev settings

benisgold commented:
idk how you guys are seeing that ui bug, i can't repro. also this is a screenshot from latest TF 

![IMG_1739](https://user-images.githubusercontent.com/15272675/184726053-6b9003d8-73da-4c02-a81b-cf8d6b28f327.PNG)

benisgold commented:
Some fixes:
![Screenshot_20220815_203539](https://user-images.githubusercontent.com/15272675/184786310-453129e7-2f49-48db-9a0a-de3e89399a61.png)
![Screenshot_20220815_203607](https://user-images.githubusercontent.com/15272675/184786324-7e010943-50c3-4912-a7ce-d3d31930523a.png)

benisgold commented:
/testflight

brunobar79 reviewed (approved):
Approving to unblock! Patch lgtm!

benisgold commented:
TF 1.7.2 (10)

will convert to SlackSheet in follow-up pr

estebanmino commented on src/components/settings-menu/AppIconSection.tsx:
```diff
@@ -88,6 +65,7 @@ const AppIconSection = () => {
   const appIconListItemsWithUnlocked = useMemo(() => {
     // Here we gotta check if each additional icon is unlocked and add it to the list
     const list = supportedAppIcons;
+    const mmkv = new MMKV();
```
curious to know why this one was moved inside the `useMemo`

estebanmino reviewed (approved):
code looking good and i went over the flows on android and ios, excepting backups
🚀

BrodyHughes reviewed (changes requested):
### QA Notes:

1. Settings > Privacy toggle doesn't seem to hide the showcase for me.
2. Settings > Privacy toggle gets stuck on one side or the other when toggled on Android. My wallet was a fresh wallet with no NFTs and no txn history, and no other wallets imported.
3. Notifications doesn't work for me still, but I think it's just a skeleton with no functionality atm.
4. Settings > Notifications not showing up for me on TF, only on my local build.
5. Wallet switcher context menu is different, but will be handled in a future PR.

benisgold commented:
@BrodyHughes 

1. it won't do anything in the app, since this it's pointless to hide your showcase from yourself (hides only to other people). you can check that it works by toggling privacy and seeing that it updates on your rainbow web profile (just verified that it works).
2. fixed! ty for the spot
3. yes, pugson is working on implementation
4. found out that feature flags don't work on TFs (maybe they can if you do something special?). i checked this by testing feature flags on unrelated TFs.
5. yep!

BrodyHughes reviewed (approved):
Ben is my hero

skylarbarrera commented:
lfg

BrodyHughes commented on src/components/secret-display/SecretDisplaySection.tsx:
```diff
@@ -0,0 +1,181 @@
+            <ToggleSecretButton onPress={loadSeed}>
+              {/* @ts-ignore */}
+              <BiometricButtonContent
+                color={colors.white}
```
On dark mode the "View secret phase button is white"

BrodyHughes replied:
was poking through the code trying to find the issue. Was wondering if this was it lol

pugson replied:
@BrodyHughes i can fix it in https://github.com/rainbow-me/rainbow/pull/4051

BrodyHughes replied:
Ohh ok. I just QA'd that and it looks good aside from this regression.

benisgold replied:
@pugson what do you think about including this? if not, nbd i got it 

https://linear.app/rainbow/issue/TEAM1-133/if-you-try-to-view-seed-phrase-and-decline-biometrics-our-explainer

pugson replied:
@benisgold sure thing i got it

BrodyHughes reviewed (commented):
@benisgold
